ec2.py

```diff
@@ -0,0 +1,119 @@
+import boto3
+import locale
+import json
+from pkg_resources import resource_filename
+
+
+# Translate between the API and what is used locally
+def translate_platform_name(operating_system, preinstalled_software):
+    os = {'Linux': 'linux', 'RHEL': 'rhel', 'SUSE': 'sles', 'Windows': 'mswin'}
+    software = {'NA': '', 'SQL Std': 'SQL', 'SQL Web': 'SQLWeb', 'SQL Ent': 'SQLEnterprise'}
+    return os[operating_system] + software[preinstalled_software]
+
+
+# Translate between the API and what is used locally
+def translate_reserved_terms(term_attributes):
+    lease_contract_length = term_attributes.get('LeaseContractLength')
+    purchase_option = term_attributes.get('PurchaseOption')
+    offering_class = term_attributes.get('OfferingClass')
+    leases = {'1yr': 'yrTerm1', '3yr': 'yrTerm3'}
+    options = {'All Upfront': 'allUpfront', 'Partial Upfront': 'partialUpfront', 'No Upfront': 'noUpfront'}
+    return leases[lease_contract_length] + str(offering_class).capitalize() + '.' + options[purchase_option]
+
+
+# The pricing API requires human readable names for some reason
+def get_region_descriptions():
+    result = dict()
+    endpoint_file = resource_filename('botocore', 'data/endpoints.json')
+    with open(endpoint_file, 'r') as f:
+        endpoints = json.load(f)
+        for partition in endpoints['partitions']:
+            for region in partition['regions']:
+                result[partition['regions'][region]['description']] = region
+    result['Asia Pacific (Osaka-Local)'] = 'ap-northeast-3'

```

Comment:
Why is this Osaka region needed like this?

Reply:
Osaka is a special region that's not accessible by general public and so it's not on the list of endpoints in boto3. (https://github.com/boto/botocore/issues/1423). However the pricing API does return data for Osaka. So either we can hardcode the list of regions or ignore Osaka. I think it can be added to the menu in another PR as I didn't want to introduce too many changes in this one.

I guess I should add a comment in code explaining it though, good catch.

Reply:
Also I'm not quite happy with getting the region descriptions in this way but I couldn't find the source for them in the source code and it was suggested in this issue: https://github.com/boto/boto3/issues/1411

Reply:
That's fine, thanks for the clarification!